homeassistant/components/amcrest/sensor.py

```diff
@@ -89,7 +89,7 @@ def available(self) -> bool:
         """Return True if entity is available."""
         return self._api.available
 
-    def update(self) -> None:
```

Comment:
```
   async def async_on_demand_update(self) -> None:
        """Update state."""
        self.async_schedule_update_ha_state(True)

    async def async_added_to_hass(self) -> None:
        """Subscribe to update signal."""
        self._unsub_dispatcher = async_dispatcher_connect(
            self.hass,
            service_signal(SERVICE_UPDATE, self._signal_name),
            self.async_on_demand_update,
        )

    async def async_will_remove_from_hass(self) -> None:
        """Disconnect from update signal."""
        assert self._unsub_dispatcher is not None
        self._unsub_dispatcher()
```

This block needs to be updated as well.

You can drop `async_on_demand_update` and call `async_write_ha_state` from the dispatcher

you can drop  `async_will_remove_from_hass` and wrap `async_dispatcher_connect` in `self.async_on_remove`

Reply:
Similar changes are needed in binary_sensor.py

Reply:
Ah, nice, thanks for the tip!